allennlp/training/trainer.py

```diff
@@ -32,7 +32,8 @@ def __init__(self,
                  cuda_device: int = -1,
                  grad_norm: Optional[float] = None,
                  grad_clipping: Optional[float] = None,
-                 log_one_line_per_batch: bool = False) -> None:
+                 log_one_line_per_batch: bool = False,
+                 validation_metric_decreases: bool = True) -> None:
```

Comment:
I don't get what you mean by "validation_metric keys which need to be matched to keys in the metric dictionary".  The key I'm suggesting doesn't change, you just check for `-` at the beginning of the `validation_metric` string, and if it's there, you remove it and set this flag.  Not a big deal, though.